### PR TITLE
Check semvar

### LINKED ISSUE #4189: Incremental release names don't follow official sumver
Tags for incremental versions don't follow Semantic Versioning. In semver, pre-release use a dash `-` e.g. `1.0.0-alpha1` or `1.0.0-rc1` while incremental release should use the plus sign e.g. `1.0.1+d2f3a45`.

Currently these tags don't meet this requirement:

* `1.9.3-6221e9ed57a10b39d363e0e97cdd6e4e41796af2`
* `v1.9.4-6277238fa1e72d2adcbab36c2a829b0eb6309533`
* `v1.9.4-abbd633e9a33877386dafd891971d6abade0ad97`

The only obstacle to using semver is that docker doesn't accept plus sign in tag name. The most common way to address this is to replace the `+` with `_` in tag names.

So the version name would be `1.9.3+6221e9ed57a10b39d363e0e97cdd6e4e41796af2` while the docker tag would be `1.9.3_6221e9ed57a10b39d363e0e97cdd6e4e41796af2`.

### PR DESCRIPTION
**Pull request by nitinjainsj**
Enforce semvar release naming.

Fixes #4189

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -99,6 +99,11 @@ point-release:
 
 # Run via 'make VERSION_ADDITIONAL=rc2 release-custom' to specify a version string
 release-candidate:
+	@if [ "$(shell echo $(VERSION_ADDITIONAL) | head -c 1)" != "-" ]; \
+	then \
+	  echo "VERSION_ADDITIONAL must be specified and start with a \"-\" character. Eg. make VERSION_ADDITIONAL=-rc2 release-custom"; \
```
Comment (ysimonson) [inline Makefile]: Super nitpick, but punctuation should be the same here vs the eg below

```suggestion
	  echo "VERSION_ADDITIONAL must be specified and start with a \"-\" character. Eg make VERSION_ADDITIONAL=-rc2 release-custom"; \
```

File: etc/build/doc [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-version="$($(GOPATH)/bin/pachctl version --client-only)"
+version="$($GOPATH/bin/pachctl version --client-only)"
```
Comment (ysimonson) [inline etc/build/doc]: Why this change? It would trigger a shellcheck warning.

Comment (nitinjainsj) [reply etc/build/doc]: I with the brackets, I get the following.

./etc/build/doc
./etc/build/doc: line 5: GOPATH: command not found
./etc/build/doc: line 5: /bin/pachctl: No such file or directory

Can you try it as well? (execute ./etc/build/doc or make VERSION_ADDITIONAL=-XYZ doc-custom) -- just to make sure something is not messed up on my side.

FWIW, this was a recent change I introduced with the last set of changes. I thought it had worked before I checked in. But since it did not work today, I am assuming I might have missed committing the right code version in the last merge.

Comment (ysimonson) [reply etc/build/doc]: Ahh that makes sense, LGTM

**Comment by nitinjainsj** [comment]
I am going to hold off on this. We still need to address docker tag issue.

**Comment by nitinjainsj** [comment]
Docker tags prevent us from being fully semvar compliant
